plugins/inputs/tacacs/tacacs_test.go

```diff
@@ -0,0 +1,353 @@
+				require.Equal(t, time.Duration(plugin.ResponseTimeout).Milliseconds(), acc.Metrics[0].Fields["responsetime_ms"])
+			}
+			if tt.name == "wrongsecret" {
+				require.ErrorContains(t, acc.Errors[0], "error on new tacacs authentication start request to "+srvLocal+" : bad secret or packet")
+			}
+			if tt.name == "unreachable" {
+				require.ErrorContains(t, acc.Errors[0], "error on new tacacs authentication start request to unreachable.hostname.com:404 : dial tcp")
+			}
+		})
+	}
+}
+
+func TestTacacsInit(t *testing.T) {
+	var testset = []struct {
+		name               string
+		testingTimeout     config.Duration
+		serverToTest       []string
+		usedUsername       config.Secret
+		usedPassword       config.Secret
+		usedSecret         config.Secret
+		requestAddr        string
+		expectNoInitErrors bool
+	}{
+		{
+			name:               "empty_creds",
+			testingTimeout:     config.Duration(time.Second * 5),
+			serverToTest:       []string{"foo.bar:80"},
+			usedUsername:       config.NewSecret([]byte(``)),
+			usedPassword:       config.NewSecret([]byte(`testpassword`)),
+			usedSecret:         config.NewSecret([]byte(`testsecret`)),
+			requestAddr:        "127.0.0.1",
+			expectNoInitErrors: false,
+		},
+		{
+			name:               "wrong_reqaddress",
+			testingTimeout:     config.Duration(time.Second * 5),
+			serverToTest:       []string{"foo.bar:80"},
+			usedUsername:       config.NewSecret([]byte(`testusername`)),
+			usedPassword:       config.NewSecret([]byte(`testpassword`)),
+			usedSecret:         config.NewSecret([]byte(`testsecret`)),
+			requestAddr:        "257.257.257.257",
+			expectNoInitErrors: false,
+		},
+		{
+			name:               "no_reqaddress",
+			testingTimeout:     config.Duration(time.Second * 5),
+			serverToTest:       []string{"foo.bar:80"},
+			usedUsername:       config.NewSecret([]byte(`testusername`)),
+			usedPassword:       config.NewSecret([]byte(`testpassword`)),
+			usedSecret:         config.NewSecret([]byte(`testsecret`)),
+			expectNoInitErrors: true,
+		},
+	}
+
+	for _, tt := range testset {
+		t.Run(tt.name, func(t *testing.T) {
+			plugin := &Tacacs{
+				ResponseTimeout: tt.testingTimeout,
+				Servers:         tt.serverToTest,
+				Username:        tt.usedUsername,
+				Password:        tt.usedPassword,
+				Secret:          tt.usedSecret,
+				RequestAddr:     tt.requestAddr,
+				Log:             testutil.Logger{},
+			}
+
+			if tt.expectNoInitErrors {
+				require.NoError(t, plugin.Init())
+			} else {
+				initErr := plugin.Init()
+				require.Error(t, initErr)
+				if tt.name == "empty_creds" {
+					require.ErrorContains(t, initErr, "empty credentials were provided (username, password or secret)")
+				}
+				if tt.name == "wrong_reqaddress" {
+					require.ErrorContains(t, initErr, "invalid ip address provided for request_ip")
+				}
+			}
+
+			if tt.name == "no_reqaddress" {
+				require.Equal(t, "127.0.0.1", plugin.RequestAddr)
+			}
```

Comment:
```suggestion
			t.plugin.Log = testutil.Logger{}
			
			err := t.plugin.Init()
			
			if tt.errContains == "" {
				require.NoError(t, err)
			} else {
				require.ErrorContains(t, err, t.errorContains)
			}
```

As it doesn't seem a best practice to have these if-cases for tests, better to have it all defined in the testset.

Reply:
Ill have a look over the coming days, also at RequireMetricEqual instead of multiple separate statements

Reply:
Removed all if-cases based on test name, now acts on testset definition